nodebuilder/rpc/config.go

```diff
@@ -44,5 +63,21 @@ func (cfg *Config) Validate() error {
 	if err != nil {
 		return fmt.Errorf("service/rpc: invalid port: %s", err.Error())
 	}
+
+	if cfg.CORS.Enabled {
+		if err := validateSecureCORS(cfg.CORS); err != nil {
+			return fmt.Errorf("service/rpc: invalid CORS config: %w", err)
+		}
+	}
+	return nil
+}
+
+// Add validation to reject wildcards
+func validateSecureCORS(config CORSConfig) error {
+	if utils.ContainsWildcard(config.AllowedOrigins) ||
+		utils.ContainsWildcard(config.AllowedMethods) ||
+		utils.ContainsWildcard(config.AllowedHeaders) {
+		return errors.New("wildcards not allowed in secure mode")
+	}
```

Comment:
Curious what is your intuition behind disallowing users to set wildcards?

Reply:
You mentioned this in the issue comment [here](https://github.com/celestiaorg/celestia-node/issues/3481#issuecomment-2866956626) that `--skip-auth` would implicitly mean wildcard for all CORS settings if I understood correctly. 

I feel the same way, but there’s a valid use case where a user might want to set only a specific CORS setting to "*".

Reply:
The validation will not allow user to set any wildcards explicitly right now.  But I think he should be able to